transaction/std/src/memo/gift_code_cancellation.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Object for 0x0202 Gift Code Cancellation memo type
+//!
+//! This was proposed for standardization in mobilecoinfoundation/mcips/pull/32
+
+use crate::{impl_memo_type_conversions, RegisteredMemoType};
+use std::convert::TryInto;
+
+/// Memo representing the cancellation of a gift code. If a gift code is
+/// never redeemed, the sender may cancel it by sending the TxOut back
+/// to their primary address. This memo will be written to the
+/// reserved change address with 8 bytes reserved for a u64 that
+/// represents the index of the cancelled gift code TxOut
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub struct GiftCodeCancellationMemo {
+    /// The data representing the gift code memo
+    memo_data: [u8; Self::MEMO_DATA_LEN],
+}
+
+impl RegisteredMemoType for GiftCodeCancellationMemo {
+    const MEMO_TYPE_BYTES: [u8; 2] = [0x02, 0x02];
+}
+
+impl GiftCodeCancellationMemo {
+    /// The length of the custom memo data
+    pub const MEMO_DATA_LEN: usize = 64;
+
+    /// Create a new gift code memo
+    pub fn new(memo_data: [u8; Self::MEMO_DATA_LEN]) -> Self {
+        GiftCodeCancellationMemo { memo_data }
+    }
+
+    /// Get the memo data
+    pub fn memo_data(&self) -> &[u8; Self::MEMO_DATA_LEN] {
+        &self.memo_data
+    }
+
+    /// Get global TxOut index of the cancelled gift code
+    pub fn cancelled_gift_code_index(&self) -> u64 {
+        u64::from_le_bytes(self.memo_data[0..8].try_into().unwrap())
+    }
+}
+
+impl From<&[u8; 64]> for GiftCodeCancellationMemo {
```

Comment:
Let's use MEMO_DATA_LEN in the type, too.
```suggestion
impl From<&[u8; GiftCodeCancellationMemo::MEMO_DATA_LEN]> for GiftCodeCancellationMemo {
```

Reply:
The difference was that I could see potentially MEMO_DATA_LEN gets shorter than 64 bytes and thus breaks the trait. I think it should be okay though unless someone wants this MEMO_DATA_LEN to be shorther.